### ISSUE TITLE
Unsupported type Type["None"]

### ISSUE THREAD

**Issue by wyfo** [python/mypy#8825]
This message is triggered by the following minimal example:
```python
from typing import Any, Type, Union

Primitive = Union[Type[str], Type[int], Type[bool], Type[float], Type[None]]  # Unsupported type Type["None"]


def primitive(data: Any, cls: Primitive):
    if cls is type(None):
        return None
    return cls(data) # The line causing the error (if I remove it, the error disappear)
```
I understand that Mypy warns that Type[None] cannot be instantiated. And, the error cannot be silenced with a `# type: ignore` on the concerned line (`return cls(data)`).
In this example, I can silence it by putting `# type: ignore` on `Primitive` declaration line. However, in my real case, the Primitive type is in a dedicated module, and the issue is even more weird, because the error is then signaled at line *1:1*. And I'm not able to silence it anymore.

Tested with Mypy 0.770 in Python 3.8.

**Comment by msullivan** [comment]
Bugs with wrong line numbers are super nasty!

**Comment by msullivan** [comment]
Easy to fix, though
